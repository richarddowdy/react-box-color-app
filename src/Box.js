import React from 'react';

function Box({ height, width, color, remove, id }){
  const handleRemove = () => {
    remove(id);
  };

  return (
    <div data-testid="box" onClick={handleRemove}>
      <div style={{ height: `${height}px`, width: `${width}px`, backgroundColor: color }}></div>
      {/* <button onClick={ handleRemove }>X</button> */}
    </div>
  )
}

export default Box;