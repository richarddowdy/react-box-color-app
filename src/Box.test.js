import React from "react";
import { render } from "@testing-library/react";
import Box from "./Box";

it("renders without crashing", function () {
  render(<Box height={100} width={100} color={'blue'}/>);
});

it("matches snapshot", function () {
  const { asFragment } = render(<Box height={100} width={100} color={'blue'}/>);
  expect(asFragment()).toMatchSnapshot();
});