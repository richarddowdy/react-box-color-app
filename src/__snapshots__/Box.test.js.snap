// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    data-testid="box"
  >
    <div
      style="height: 100px; width: 100px; background-color: blue;"
    />
  </div>
</DocumentFragment>
`;
